corehq/apps/userreports/tests/test_expressions.py

```diff
@@ -743,13 +743,13 @@ def test_caching(self):
         self.test_simple_lookup()
 
         my_doc = self.database.get('my-id')
-        self.database.mock_docs.clear()
 
-        self.assertEqual({}, self.database.mock_docs)
-        self.assertEqual('foo', self.expression(my_doc, EvaluationContext(my_doc, 0)))
+        context = EvaluationContext(my_doc, 0)
+        self.assertEqual('foo', self.expression(my_doc, context))
 
-        same_expression = ExpressionFactory.from_spec(self.spec)
-        self.assertEqual('foo', same_expression(my_doc, EvaluationContext(my_doc, 0)))
+        my_doc = self.database.get('my-id')
+        self.database.mock_docs.clear()
+        self.assertEqual('foo', self.expression(my_doc, context))
```

Comment:
yes, if you remove the caching portion the context tries to fetch the document from the database and returns an empty result. if this line were to change to `self.assertEqual('foo', self.expression(my_doc, EvaluationContext(my_doc, 0)))` it'd fail